bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/DraftsAndSubmissionsImpl.java

```diff
@@ -0,0 +1,358 @@
+            .stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        return operation.execute(inputMap);
+    }
+
+    @Override
+    public String getOperationName() {
+        String operationParam = request.getParameter(OPERATION);
+        return operationParam;
+    }
+
+    @Override
+    protected List<PortalLister.Item> getItemList() {
+        List<PortalLister.Item> itemList = new ArrayList<>();
+        TypeEnum typeEnum = TypeEnum.valueOf(getType());
+
+        QueryImpl query = new QueryImpl();
+        query.setOffset(getOffset());
+        query.setLimit(getLimit());
+
+        // Add information about currently logged in user
+        StatementImpl currentUserStatement = new StatementImpl();
+        currentUserStatement.setAttributeName("owner");
+        currentUserStatement.setAttributeValue(resolver.getUserID());
+        currentUserStatement.setOperator(Statement.Operator.EQUALS);
+        StatementGroupImpl stmtGroup = new StatementGroupImpl();
+        stmtGroup.addStatement(currentUserStatement);
+        query.setStatementGroup(stmtGroup);
+
+        switch (typeEnum) {
+            case DRAFT:
+                try {
+                    List<DraftModel> list = draftService.getAllDraft(query);
+                    for (DraftModel draftModel : list) {
+                        PortalLister.Item item = getItem(draftModel.getFormPath(), typeEnum, draftModel.getId(),
+                            draftModel.getLastModifiedTime().getTimeInMillis() + "");
+                        itemList.add(item);
+                    }
+                } catch (FormsPortalException e) {
+                    LOGGER.error("Failed to fetch Form Drafts.", e);
+                }
+                break;
+            case SUBMISSION:
+                try {
+                    List<SubmitModel> list = submitService.getAllSubmission(query);
+                    for (SubmitModel submitModel : list) {
+                        PortalLister.Item item = getItem(submitModel.getFormPath(), typeEnum, submitModel.getId(),
+                            submitModel.getLastModifiedTime().getTimeInMillis() + "");
+                        itemList.add(item);
+                    }
+                } catch (FormsPortalException e) {
+                    LOGGER.error("Failed to fetch Form Submissions.", e);
+                }
+                break;
+        }
+
+        // might create holes during pagination
+        return itemList.stream().filter(Objects::nonNull).collect(Collectors.toList());
+    }
+
+    @Override
+    public Map<String, Object> getElements() {
+        if (isOperationCall()) {
+            return null;
+        }
+        return super.getElements();
+    }
+
+    @Override
+    public String getTitle() {
+        if (isOperationCall()) {
+            return null;
+        }
+        return super.getTitle();
+    }
+
+    @Override
+    public String getLayout() {
+        if (isOperationCall()) {
```

Comment:
This does not look good to me. Looks like a side effect, function is trying to do something which it is not intended to do. Can we fix this

Reply:
Only other way I can think is to create a different servlet (with a different selector) for performing actions.

cc @sakshi-arora1

Reply:
Created servlet